lib/Auth/Process/PerunEntitlement.php

```diff
@@ -0,0 +1,201 @@
+    private $groupNamePrefix;
+    private $groupNameAARC;
+    private $groupNameAuthority;
+    private $interface;
+    private $adapter;
+
+    public function __construct($config, $reserved)
+    {
+        parent::__construct($config, $reserved);
+        $conf = Configuration::getConfig(self::CONFIG_FILE_NAME);
+        assert('is_array($config)');
+
+        if (!isset($config[self::EDU_PERSON_ENTITLEMENT])) {
+            throw new Exception(
+                'perun:PerunEntitlement: missing mandatory configuration option ' .
+                self::EDU_PERSON_ENTITLEMENT . '.'
+            );
+        }
+        $this->eduPersonEntitlement = $config[self::EDU_PERSON_ENTITLEMENT];
+
+        if (!isset($config[self::FORWARDED_EDU_PERSON_ENTITLEMENT])) {
+            throw new Exception(
+                'perun:PerunEntitlement: missing mandatory configuration option ' .
+                self::FORWARDED_EDU_PERSON_ENTITLEMENT . '.'
+            );
+        }
+        $this->forwardedEduPersonEntitlement = $config[self::FORWARDED_EDU_PERSON_ENTITLEMENT];
+
+        $this->groupNamePrefix = $conf->getString(self::GROUPNAMEPREFIX_ATTR, '');
+        $this->groupNameAuthority = $conf->getString(self::GROUPNAMEAUTHORITY_ATTR, '');
+        $this->groupNameAARC = $conf->getBoolean(self::GROUPNAMEAARC_ATTR, false);
+
+        if ($this->groupNameAARC && (empty($this->groupNameAuthority) || empty($this->groupNamePrefix))) {
+            throw new Exception(
+                'perun:PerunEntitlement: \'groupNameAARC\' has been set, \'groupNameAuthority\' ' .
+                'and \'groupNamePrefix\' options must be set as well'
+            );
+        }
+
+        if (!isset($config[self::INTERFACE_PROPNAME])) {
+            $config[self::INTERFACE_PROPNAME] = Adapter::RPC;
+        }
+
+        $this->interface = (string)$config[self::INTERFACE_PROPNAME];
+        $this->adapter = Adapter::getInstance($this->interface);
+    }
+
+    public function process(&$request)
+    {
+        $eduPersonEntitlement = $this->getEduPersonEntitlement($request);
+        $forwardedEduPersonEntitlement = $this->getForwardedEduPersonEntitlement($request);
+        $resourceCapabilities = $this->getResourceCapabilities($request);
+
+        $request['Attributes'][$this->eduPersonEntitlement] = array_merge(
+            $eduPersonEntitlement,
+            $forwardedEduPersonEntitlement,
+            $resourceCapabilities
+        );
+    }
+
+    private function getEduPersonEntitlement(&$request)
+    {
+        if (isset($request['perun']['groups'])) {
+            /** allow IDE hint whisperer
+             * @var model\Group[] $groups
+             */
+            $groups = $request['perun']['groups'];
+        } else {
+            throw new Exception(
+                'perun:PerunEntitlement: ' .
+                'missing mandatory field \'perun.groups\' in request.' .
+                'Hint: Did you configured PerunIdentity filter before this filter?'
```

Comment:
Using _did_ removes the _ed_ suffix from _configured_ :). See https://learnenglish.britishcouncil.org/english-grammar-reference/past-simple